File: atlaskv-ts-sdk/src/models/ClusterStatus.ts
/**
 * Represents the current status and consensus details of an AtlasKV cluster node.
 */
export interface ClusterStatus {
  /** The unique ID of the queried node. */
  nodeId: string;
  /** The current Raft role of the node (e.g., LEADER, FOLLOWER, CANDIDATE). */
  role: string;
  /** The current consensus term number. */
  currentTerm: number;
  /** The highest committed log index. */
  commitIndex: number;
  /** The highest applied log index in the state machine. */
  lastApplied: number;
  /** The node ID of the current leader, or null if unknown. */
  currentLeader: string | null;
  /** Indicates whether the node is healthy and operational. */
  healthy: boolean;
  /** The node's uptime in milliseconds. */
  uptimeMs: number;
  /** The lifecycle state of the node (e.g., STARTED, STOPPING). */
  nodeState: string;
  /** The port on which the node's internal gRPC engine is listening. */
  grpcPort: number;
  /** The number of configured peers in the cluster. */
  peerCount: number;
}

File: atlaskv-java-sdk/src/main/java/com/atlaskv/sdk/models/ClusterStatus.java
package com.atlaskv.sdk.models;

/**
 * Immutable cluster status model.
 *
 * @param nodeId        identity of the queried node
 * @param role          current Raft role (LEADER, FOLLOWER, CANDIDATE)
 * @param currentTerm   current Raft consensus term
 * @param commitIndex   highest committed log index
 * @param lastApplied   highest applied log index
 * @param currentLeader current leader node ID (null if unknown)
 * @param healthy       true if the node is operational
 * @param uptimeMs      uptime in milliseconds
 * @param nodeState     lifecycle state of the node (e.g. STARTED, STOPPING)
 * @param grpcPort      gRPC server listening port
 * @param peerCount     number of configured peers in the cluster
 */
public record ClusterStatus(
        String nodeId,
        String role,
        long currentTerm,
        long commitIndex,
        long lastApplied,
        String currentLeader,
        boolean healthy,
        long uptimeMs,
        String nodeState,
        int grpcPort,
        int peerCount
) {}
